File: 151003/Matoshko/tweet-services/tweet-service-discussion/src/main/java/service/tweetservicediscussion/serivces/MessageService.java
package service.tweetservicediscussion.serivces;

import jakarta.validation.Valid;
import org.springframework.validation.annotation.Validated;
import service.tweetservicediscussion.domain.entity.ValidationMarker;
import service.tweetservicediscussion.domain.request.MessageRequestTo;
import service.tweetservicediscussion.domain.response.MessageResponseTo;

import java.util.List;

public interface MessageService {
    @Validated(ValidationMarker.OnCreate.class)
    MessageResponseTo create(@Valid MessageRequestTo entity);

    List<MessageResponseTo> read();

    @Validated(ValidationMarker.OnUpdate.class)
    MessageResponseTo update(@Valid MessageRequestTo entity);

    Long delete(Long id);

    MessageResponseTo findMessageById(Long id);
}

File: 151004/Burbouski/appLab1/src/main/kotlin/com/github/hummel/dc/lab1/service/MessageService.kt
package com.github.hummel.dc.lab1.service

import com.github.hummel.dc.lab1.dto.request.MessageRequestTo
import com.github.hummel.dc.lab1.dto.request.MessageRequestToId
import com.github.hummel.dc.lab1.dto.response.MessageResponseTo

interface MessageService {
	suspend fun getAll(): List<MessageResponseTo>

	suspend fun create(messageRequestTo: MessageRequestTo?): MessageResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getById(id: Long): MessageResponseTo?

	suspend fun update(messageRequestToId: MessageRequestToId?): MessageResponseTo?
}